modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -418,16 +419,31 @@
          driver
          database
          nil
+         ;; you know what, if we really wanted to make this efficient we would do
+         ;;
+         ;;    SHOW SCHEMAS IN DATABASE <db>
+         ;;
+         ;; first, and filter out the ones we're not interested in, and THEN do
+         ;;
+         ;;    SHOW OBJECTS IN SCHEMA <schema>
+         ;;
+         ;; for each of the schemas we wanted to sync. Right now we're fetching EVERY table, including ones from schemas
+         ;; we aren't interested in.
          (fn [^Connection conn]
-           {:tables (->> (sql-jdbc.describe-database/db-tables driver (.getMetaData conn) nil db-name)
-                         (filter (fn [{schema :schema table-name :name}]
-                                   (and (not (contains? excluded-schemas schema))
-                                        (driver.s/include-schema? inclusion-patterns
-                                                                  exclusion-patterns
-                                                                  schema)
-                                        (sql-jdbc.sync/have-select-privilege? driver conn schema table-name))))
-                         (map #(dissoc % :type))
-                         set)}))))))
+           {:tables (into #{}
```

Comment:
The only thing I actually changed here was changing the schema pattern from `nil` to `"%"`, but I also reworked this to transduce rather than do things with lazy seqs for the slight performance boost and because the old sync code did stuff that way